src/main/java/com/tabnine/general/GettingStartedManager.kt

```diff
@@ -0,0 +1,92 @@
+package com.tabnine.general
+
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx
+import com.intellij.openapi.fileEditor.impl.HTMLEditorProvider
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.tabnine.general.Utils.executeUIThreadWithDelay
+import java.lang.reflect.Method
+import java.util.Arrays
+import java.util.concurrent.TimeUnit
+import javax.swing.SwingConstants
+
+const val PAGE_TITLE = "Tabnine - Getting Started"
+const val PAGE_URL = "https://www.tabnine.com/getting-started/intellij?origin=ide"
+const val IS_GETTING_STARTED_OPENED_KEY = "is-getting-started-opened"
+const val AFTER_PROJECT_OPENED_DELAY_SECONDS = 15L
+
+fun handleFirstTimePreview() {
+    if (!isPageShown()) {
+        executeUIThreadWithDelay(
+            { openPageOnAllProjects() },
+            AFTER_PROJECT_OPENED_DELAY_SECONDS,
+            TimeUnit.SECONDS
+        )
+    }
+}
+
+fun openPageOnAllProjects() {
+    if (isInIdeWebPageSupported()) {
```

Comment:
no need for this `if` now that you `try/catch` inside the function

Reply:
ok. I just thought it will be clearer that way..